src/plugins/telemetry/server/config/telemetry_labels.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+
+/**
+ * Labels to enrich the context of the telemetry generated.
+ * When adding new keys, bear in mind that this info is exposed
+ * to the browser **even to unauthenticated pages**.
+ */
+export const labelsSchema = schema.object(
```

Comment:
Mind that I explicitly didn't add the `kibana_uuid` because it's already part of the context (server-side only). Reporting it public-side does not make much sense to me because the browser can hit multiple instances of Kibana.

Reply:
Yes definitely 👍🏽 , as far as I can see we need all other fields reported explicitly since they do not belong to context. And also as you mentioned `branch` info is always **main** we would wanna get it from env rather than package.json